Update third_party/renode/README.md (#231)
iml130/iree-bare-metal-arm: third_party/renode/README.md

- The file [`stm32f446.repl`](https://github.com/renode/renode/blob/381e0c59bd97a1e5f5348222bbeeec5588a68575/platforms/cpus/stm32f4.repl) (renamed from `stm32f4.repl`) was imported at state renode/renode@381e0c5. It is modified for the STM32F446 according to the [`datasheet`](https://www.st.com/resource/en/datasheet/stm32f446re.pdf) and [`reference manual`](https://www.st.com/resource/en/reference_manual/rm0390-stm32f446xx-advanced-armbased-32bit-mcus-stmicroelectronics.pdf).

Further, we provide files, which use upstream platform description files from [renode/renode](https://github.com/renode/renode/) and modify specific definitions for the needs of the iree-bare-metal-arm project.

- The file `nrf52840.repl` modifies the definition of `uart0` to not use DMA.
- The file `stm32f746.repl` modifies flash and SRAM for the STM32F746 according to the [`datasheet`](https://www.st.com/resource/en/datasheet/stm32f746zg.pdf) and [`reference manual`](https://www.st.com/resource/en/reference_manual/rm0385-stm32f75xxx-and-stm32f74xxx-advanced-armbased-32bit-mcus-stmicroelectronics.pdf).

> In addition to that we provide files originally created in the context of the iree-bare-metal-arm project.

Reviewer: ```suggestion
Moreover, we provide additional files which were specifically created in the context of the iree-bare-metal-arm project.
```
